frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -141,7 +145,9 @@ export type PrometheusResponse = {
 };
 
 export type WatchK8sResource = {
-  kind: K8sResourceKindReference;
+  /** @deprecated use groupVersionKind instead, as migrating towards usage a group/version/kind object instead of the '~' concat stings */
```

Comment:
I'd say something like

```suggestion
  /** @deprecated Use groupVersionKind instead. The kind property will be removed in a future release. */
```